tests/io/test_multi_table.py

```diff
@@ -0,0 +1,242 @@
+from pathlib import Path
+
+import pytest
+from anndata import AnnData
+from anndata.tests.helpers import assert_equal
+from spatialdata import SpatialData
+from spatialdata.models import TableModel
+
+from tests.conftest import _get_shapes, _get_table
+
+# notes on paths: https://github.com/orgs/scverse/projects/17/views/1?pane=issue&itemId=44066734
+# notes for the people (to prettify) https://hackmd.io/wd7K4Eg1SlykKVN-nOP44w
+test_shapes = _get_shapes()
+
+# shuffle the indices of the dataframe
+# np.random.default_rng().shuffle(test_shapes["poly"].index)
+
+
+class TestMultiTable:
+    def test_set_get_tables_from_spatialdata(self, full_sdata: SpatialData, tmp_path: str):
+        tmpdir = Path(tmp_path) / "tmp.zarr"
+        adata0 = _get_table(region="polygon")
+        adata1 = _get_table(region="multipolygon")
+        full_sdata["adata0"] = adata0
+        full_sdata["adata1"] = adata1
+
+        adata2 = adata0.copy()
+        del adata2.obs["region"]
+        with pytest.raises(ValueError):
+            full_sdata["not_added_table"] = adata2
+
+        assert len(full_sdata.tables) == 3
+        assert "adata0" in full_sdata.tables and "adata1" in full_sdata.tables
+        full_sdata.write(tmpdir)
+
+        full_sdata = SpatialData.read(tmpdir)
+        assert_equal(adata0, full_sdata["adata0"])
+        assert_equal(adata1, full_sdata["adata1"])
+        assert "adata0" in full_sdata.tables and "adata1" in full_sdata.tables
+
+    @pytest.mark.parametrize(
+        "region_key, instance_key, error_msg",
+        [
+            (None, None, "Instance key and/or instance"),
+            ("region", None, "Instance key and/or instance"),
+            ("region", "instance_id", "Instance key instance_id not"),
+            (None, "instance_id", "Instance key instance_id not"),
+        ],
+    )
+    def test_change_annotation_target(self, full_sdata, region_key, instance_key, error_msg):
+        n_obs = full_sdata["table"].n_obs
+        with pytest.raises(ValueError, match="Mismatch found between regions"):
+            full_sdata.set_table_annotation_target("table", "poly")
```

Comment:
This is purely based on a set operation. I wrote it now in a join to have it look more neat, but I think it would be too much to also provide the element type as this is not of particular interest.

Reply:
ah ok you are right, `poly` was a elements name not the element type, I read it wrong